Makefile

```diff
@@ -28,42 +45,66 @@ buildbox:
 		docker/build
 
 .PHONY: containers
+containers: ## Build container images.
 containers: build lint
 	$(MAKE) -C docker containers DOCKER_ARGS=$(DOCKER_ARGS)
 
 .PHONY: publish
+publish: ## Publish container images to quay.io.
 publish: build lint
-	$(MAKE) -C docker -j publish TAG=$(TAG)
-
-#
-# Runs inside build container
-#
-
-.PHONY: $(TARGETS)
-$(TARGETS): vendor
-	@go version
-	cd $(SRCDIR) && \
-		GO111MODULE=on go test -mod=vendor -c -i ./$(subst robotest-,,$@) -o build/robotest-$@
-
-vendor: go.mod
-	cd $(SRCDIR) && go mod vendor
+	$(MAKE) -C docker -j publish VERSION=$(DOCKER_VERSION) TAG=$(TAG)
 
 .PHONY: clean
-clean:
+clean: ## Remove intermediate build artifacts & cache.
 	@rm -rf $(BUILDDIR)/*
 	@rm -rf vendor
 
 .PHONY: test
-test:
+test: ## Run unit tests.
 	docker run $(DOCKERFLAGS) \
 		--env="GO111MODULE=off" \
 		$(BUILDBOX) \
 		dumb-init go test -cover -race -v ./infra/...
 
 .PHONY: lint
+lint: ## Run static analysis against source code.
 lint: buildbox
 	docker run $(DOCKERFLAGS) \
 		--env="GO111MODULE=off" \
 		$(BUILDBOX) dumb-init golangci-lint run \
 		--skip-dirs=vendor \
 		--timeout=2m
+
+.PHONY: version
+version: ## Show the robotest version.
+version: $(VERSIONFILE)
```

Comment:
I'm not sure investing in Makefile templating is worth it. We've been gradually migrating to [mage](https://github.com/magefile/mage) - some examples are [satellite](https://github.com/gravitational/satellite/blob/master/build.go), [planet](https://github.com/gravitational/planet/blob/master/build.go) and [wormhole](https://github.com/gravitational/wormhole/blob/master/build.go).

Reply:
I wasn't particularly fond of this templating either.  Its something I copied from https://github.com/gravitational/teleport/blob/8c75759b98fbf71f2932c6f605c7d8feb4e50dfd/version.mk as the least complex way of dynamically generating versions that I've found.

I'll take a look at mage.  I suspect that migration would considerably increase the scope of this PR though, so I may work on it as a separate work item.

Reply:
You don't need to do a complete rewrite in Mage, new targets can be setup in mage / go, but existing stuff doesn't necessarily have to change right of way. And a make target can just call into mage as well as a stop gap.

Reply:
I took an hour yesterday to POC a target or two with mage.  I estimate maybe two full days of my time to migrate over to it (one for conversion, a 2nd for testing, jenkins cleanup & maybe PR builds #171).  As such I've filed https://github.com/gravitational/robotest/issues/207 for mage work.

> You don't need to do a complete rewrite in Mage, new targets can be setup in mage / go, but existing stuff doesn't necessarily have to change right of way.

Given the amount of work, I'd prefer not to leave things in a partially migrated state.  I will make the time for a full conversion before 2.0 (which I'm aiming for at the end of the month).

The remaining question is: Should we take the work in this PR, or scrap this until mage arrives?  I'm inclined to take it as the 80 lines here:

1) are largely consistent with how our other projects manage versions
2) won't make the mage migration significantly harder
3) provide some immediate value (specifically, it allows me to easily figure out which git commit a robotest container in the wild corresponds to)